### PR TITLE
Support multi-qubit measurements in deferred measurement transformer

### LINKED ISSUE #5694: Multi-qubit conditions in DeferredMeasurementsTransformer
Use https://github.com/quantumlib/Cirq/pull/5755 to allow control keys from multi-qubit measurements in deferred measurements transformers. That new feature allows us to express "anything but all zeros", which is the condition required.

https://github.com/quantumlib/Cirq/blob/77b61f500af3726ad4b34b06c72e21eac48d3db6/cirq-core/cirq/transformers/measurement_transformers.py#L121-L124

### PR DESCRIPTION
**Pull request by daxfohl**
Use SumOfProducts to enable controls from multi-qubit measurements in deferred measurement transformer.

Fixes #5694

### CODE REVIEW

File: cirq-core/cirq/transformers/measurement_transformers.py [hunk 1]
```diff
@@ -111,23 +110,22 @@ def defer(op: 'cirq.Operation', _) -> 'cirq.OP_TREE':
...
-                        # Remember to update docstring above once this works.
-                        raise ValueError('Only single qubit conditions are allowed.')
-                    controls.extend(qubits)
+                    qs = measurement_qubits[c.key]
+                    if len(qs) == 1:
+                        control_values: Any = range(1, qs[0].dimension)
+                    else:
+                        all_values = itertools.product(*[range(q.dimension) for q in qs])
+                        anything_but_all_zeros = tuple(itertools.islice(all_values, 1, None))
+                        control_values = ops.SumOfProducts(anything_but_all_zeros)
```
Comment (tanujkhattar) [inline cirq-core/cirq/transformers/measurement_transformers.py:124]:
```suggestion
                    all_values = itertools.product(*[range(q.dimension) for q in qs])
                    anything_but_all_zeros = tuple(itertools.islice(all_values, 1, None))
                    control_values = ops.SumOfProducts(anything_but_all_zeros)
```

**Review by tanujkhattar — CHANGES_REQUESTED** [review]
Are there any plans to provide a more explicit control on key conditions for a classically controlled operation? For example, can we say "Execute X(q0) only if measurement result of q1 and q0 is either both 0 or both 1" ? I guess this would require extending classically controlled operations to also accept control values? 

It was a bit weird to see that the default strategy we are following for multi qubit classical controls is to activate the classical control on "anything but zeros". This can probably be a good first start, but I'm curious to hear if you've already thought on this.

File: cirq-core/cirq/transformers/measurement_transformers.py (hunk 1 above)
Comment (daxfohl) [reply cirq-core/cirq/transformers/measurement_transformers.py:124]: Logically yes, but for a single qubit, the range gets read as a ProductOfSums, which prints nicer circuit diagrams. (Last I checked SumOfProducts failed to print at all).

Comment (tanujkhattar) [reply cirq-core/cirq/transformers/measurement_transformers.py:124]: After https://github.com/quantumlib/Cirq/pull/5788 is merged; `SumOfProducts` will also have nice circuit diagrams.

Comment (tanujkhattar) [reply cirq-core/cirq/transformers/measurement_transformers.py:124]: Hmm, I think this is fine.

**Review by tanujkhattar — APPROVED** [review]

**Pull request merged**